tests/core/dom/range/extractcontents.js

```diff
@@ -379,6 +379,38 @@
 
 			// See: execContentsAction in range.js.
 			assert.isInnerHtmlMatching( '<p>Foo bar</p>', docFrag.getHtml(), 'Extracted HTML' );
+		},
+
+		// #644
+		'test extract nested tags': function() {
+			bender.editorBot.create( {
```

Comment:
I don't think there's a need to create a dedicated editor here.

Instead move `<p>111<strong>222<span>333<em>444</em></span>555<sup>666</sup>777<sub>888</sub></strong></p>` fixture to the `.html` file and use it here (with a proper selection).